srml/executive/src/lib.rs

```diff
@@ -235,7 +253,7 @@ impl<
 			return Err(internal::ApplyError::FullBlock);
 		}
 
-		if let (Some(sender), Some(index)) = (xt.sender(), xt.index()) {
+		if let (Some(true), Some(sender), Some(index)) = (uxt_signature, xt.sender(), xt.index()) {
```

Comment:
this shouldn't be necessary: `xt.sender()` can only be `Some` if there's a signature (`sender` is the signature's signer).